src/NextGen/DonationForm/Actions/ConvertDonationFormBlocksToFieldsApi.php

```diff
@@ -146,6 +148,73 @@ protected function createNodeFromDonorNameBlock(BlockModel $block): Node
         });
     }
 
+    /**
+     * @unreleased
+     */
+    protected function createNodeFromAmountBlock(BlockModel $block): Node
+    {
+        return DonationAmount::make('donationAmount')->tap(function (Group $group) use ($block) {
+            $amountRules = ['required', 'numeric'];
+
+            if ($block->hasAttribute('customAmount') && $block->getAttribute('customAmount') === true) {
+                if ($block->hasAttribute('customAmountMin')) {
+                    $amountRules[] = "min:{$block->getAttribute('customAmountMin')}";
+                }
+
+                if ($block->hasAttribute('customAmountMax') && $block->getAttribute('customAmountMax') > 0) {
+                    $amountRules[] = "max:{$block->getAttribute('customAmountMax')}";
+                }
+            }
+
+            /** @var Amount $amount */
+            $amount = $group->getNodeByName('amount');
+            $amount
+                ->label(__('Donation Amount', 'give'))
+                ->attributes($block->attributes)
+                ->levels(...array_map('absint', $block->attributes['levels']))
+                ->allowLevels($block->attributes['priceOption'] === 'multi')
+                ->allowCustomAmount($block->attributes['customAmount'])
+                ->fixedAmountValue($block->hasAttribute('setPrice') ? $block->attributes['setPrice'] : null)
+                ->defaultValue(
+                    $block->attributes['priceOption'] === 'set' && $block->hasAttribute(
+                        'setPrice'
+                    ) ? $block->attributes['setPrice'] : 50
+                )
+                ->rules(...$amountRules);
+
+            /** @var Hidden $currency */
+            $currency = $group->getNodeByName('currency');
+            $currency
+                ->defaultValue(give_get_currency($this->formId))
```

Comment:
I would suggest hoisting this call to `give_get_currency()` and setting a `$currency` property via the constructor. This would help to decouple the process of the `createNodeFromAmountBlock()` method.

Reply:
resolved 6f2f7f3d